test/e2e/resize_nodes.go

```diff
@@ -400,12 +439,30 @@ var _ = Describe("Nodes", func() {
 			waitForNodeToBe(c, node.Name, true, 2*time.Minute)
 
 			By("verify wheter new pods can be created on the re-attached node")
-			err = resizeReplicationController(c, ns, name, replicas+1)
-			Expect(err).NotTo(HaveOccurred())
-			err = waitForPodsCreatedRunningResponding(c, ns, name, replicas+1)
-			Expect(err).NotTo(HaveOccurred())
-			_, err = waitForRCPodOnNode(c, ns, name, node.Name)
-			Expect(err).NotTo(HaveOccurred())
+			additionalPod := "additionalpod"
+			expectNoError(createServeHostnamePodOnNode(c, ns, additionalPod, node.Name))
+			expectNoError(waitForPodsCreatedRunningResponding(c, ns, additionalPod, 1))
+
+			// verify that it is really on the requested node
+			{
+				pod, err := c.Pods(ns).Get(additionalPod)
+				Expect(err).NotTo(HaveOccurred())
+				if pod.Spec.NodeName != node.Name {
+					Logf("Pod %s found on invalid node: %s instead of %s", pod.Spec.NodeName, node.Name)
+				}
+			}
+
+			// an alternative way of checking if pods are scheduled on the recovered
+			// node would be to increase the RC size by 1 and check that the new
+			// pod is scheduled on the recovered node. Something like:
+			// * resizeReplicationController(c, ns, name, replicas+1)
+			// * waitForPodsCreatedRunningResponding(c, ns, name, replicas+1)
+			// * waitForRCPodOnNode(c, ns, name, node.Name)
+			//
+			// But the problem with this approach is that we rely on the scheduler
+			// implementation that may change. For instance, if rescheduling is added,
+			// the pods will go back to original nodes and the new pod may be scheduled
+			// anywhere in the cluster, not only on the recovered node.
```

Comment:
Suggest removing this comment block, or phrasing as "increasing the RC size is not a valid way to test this because...", since it *isn't* an alternative way of doing this :)